packages/middleware-flexible-checksums/src/middleware-flexible-checksums.integ.spec.ts

```diff
@@ -80,5 +82,35 @@ describe("middleware-flexible-checksums", () => {
 
       expect.hasAssertions();
     });
+
+    it("should not set binary file content length", async () => {
+      const client = new S3({ region: "us-west-2" });
+
+      requireRequestsFrom(client).toMatch({
+        method: "PUT",
+        hostname: "s3.us-west-2.amazonaws.com",
+        protocol: "https:",
+        path: "/b/k",
+        headers: {
+          "content-type": "application/octet-stream",
+          "x-amz-content-sha256": "STREAMING-UNSIGNED-PAYLOAD-TRAILER",
+          "content-length": /undefined/,
+        },
+        query: {
+          "x-id": "PutObject",
+        },
+      });
+
+      const stream = fs.createReadStream(path.join(__dirname, "..", "test", "green.jpeg"));
```

Comment:
(minor) Could this be instead be achieved by a base64-encoded representation of an image that could fit the file instead of an additional external resource?

Reply:
I'll look into doing this in a revision, I have to fix some more tests anyway.